python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -495,29 +495,33 @@ def load_targets(
                 targets.append(target.load())
         return targets
 
-    def check_valid_run_requests(self, run_requests: Sequence[RunRequest]):
+    def check_valid_run_requests(
+        self, run_requests: Sequence[RunRequest], context: SensorEvaluationContext
+    ):
         has_multiple_targets = len(self._targets) > 1
         target_names = [target.pipeline_name for target in self._targets]
 
-        if run_requests and not self._targets and not self._asset_selection:
-            raise Exception(
-                f"Error in sensor {self._name}: Sensor evaluation function returned a RunRequest "
-                "for a sensor lacking a specified target (job_name, job, or jobs). Targets "
-                "can be specified by providing job, jobs, or job_name to the @sensor "
-                "decorator."
-            )
-
         for run_request in run_requests:
             if run_request.job_name is None and has_multiple_targets:
                 raise Exception(
                     f"Error in sensor {self._name}: Sensor returned a RunRequest that did not "
                     f"specify job_name for the requested run. Expected one of: {target_names}"
                 )
-            elif run_request.job_name and run_request.job_name not in target_names:
+            elif run_request.job_name and target_names and run_request.job_name not in target_names:
                 raise Exception(
                     f"Error in sensor {self._name}: Sensor returned a RunRequest with job_name "
                     f"{run_request.job_name}. Expected one of: {target_names}"
                 )
+            elif (
+                run_request.job_name
+                and context.repository_def
+                and not context.repository_def.has_job(run_request.job_name)
+                and not context.repository_def.has_pipeline(run_request.job_name)
```

Comment:
nit: you can drop pipeline since we no longer support them

Reply:
Heh my test failed if I left this out - I will update the test to use a job and take this out

Reply:
I left this in b/c it seems like you can still write code that hits it?